internal/convert/converter_test.go

```diff
@@ -896,13 +902,85 @@ func TestConverter(t *testing.T) { //nolint:maintidx
 			to:          pointer(OuterStruct{}),
 			err:         "'' expected a map, got 'string'",
 		},
-		// unsupported.
 		{
-			description: "to interface (unsupported)",
+			description: "int to interface",
+			from:        42,
+			to:          pointer(any(nil)),
+			expected:    pointer(any(42)),
+		},
+		{
+			description: "string to interface",
 			from:        "str",
 			to:          pointer(any(nil)),
-			err:         ": unsupported type: interface",
+			expected:    pointer(any("str")),
+		},
+		{
+			description: "float to interface",
+			from:        42.42,
+			to:          pointer(any(nil)),
+			expected:    pointer(any(42.42)),
 		},
+		{
+			description: "map to interface",
```

Comment:
Added few more tests to check if maps.Unpack call is required. Seems to be excessive.

Reply:
I may miss test for Unpack in convert package. Here is the code calls maps.Unpack https://github.com/nil-go/konf/blob/857c68e060f9e24b961d1abaff28175c91969461/internal/convert/converter.go#L355

Reply:
Just tested it from most outer https://github.com/nil-go/konf/blob/main/config_test.go#L102

Reply:
> I may miss test for Unpack in convert package. Here is the code calls maps.Unpack
> 
> https://github.com/nil-go/konf/blob/857c68e060f9e24b961d1abaff28175c91969461/internal/convert/converter.go#L355

Seems to be tested indirectly.